napari/components/_layer_slicer.py

```diff
@@ -68,9 +71,46 @@ def __init__(self):
         """
         self.events = EmitterGroup(source=self, ready=Event)
         self._executor: Executor = ThreadPoolExecutor(max_workers=1)
+        self._force_sync = False
         self._layers_to_task: Dict[Tuple[Layer], Future] = {}
         self._lock_layers_to_task = RLock()
 
+    @contextmanager
+    def force_sync(self):
+        """Context manager to temporarily force slicing to be synchronous.
+        This should only be used from the main thread.
+
+        >>> layer_slicer = _LayerSlicer()
+        >>> layer = Image(...)  # an async-ready layer
+        >>> with layer_slice.force_sync():
+        >>>     layer_slicer.slice_layers_async(layers=[layer], dims=Dims())
+        """
+        self._force_sync = True
+        yield None
+        self._force_sync = False
```

Comment:
```suggestion
        prev = self._force_sync
        self._force_sync = True
        try:
            yield None
        finally:
            self._force_sync = prev
```
https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager